### PR TITLE
[Torch] Canonicalize aten.convolution with single int tuple params

### PR DESCRIPTION
**Pull request by hariprasadravi**
This PR is a more robust fix for the issue captured in #4380 .

Essentially, `torch.ops.aten.convolution` lowering to `tosa`, `linalg` and `stablehlo` fail if `stride`, `padding`, `dilation` or `output_padding` is a tuple with a singleton element, but convolution operates on 2 or 3 spatial dimensions.

In the failing case, `torch.nn.Conv2d` with `padding= 'valid'` generates a `torch.ops.aten.conv2d.padding` op in `ExportedProgram`, which is later decomposed to `torch.ops.aten.convolution.default` with a single padding value of [0] after running `ep.run_decompositions()`. 

In #4380, I attempted to fix just the `torch-to-tosa` pass, but I later realised that this is a more general bug in all the backends for multiple params (thanks to https://github.com/llvm/torch-mlir/pull/4380#discussion_r2542431544).

### Fix:
I followed #4250 to canonicalize aten.convolution if it operates on 2 or 3 spatial dims but params are singleton. For example, if `aten.convolution` is 2D but `padding == [0]`, we canonicalize it to `padding == [0, 0]`

### CODE REVIEW

**Comment by hariprasadravi** [comment]
@sahas3 @sjarus @zjgarvey and @vivekkhandelwal1 . Please let me know if you have any thoughts.

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -5898,6 +5899,160 @@ void AtenMaxPool3dOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
...
+    }
+
+    SmallVector<int64_t, 3> outputPaddingInts;
+    if (!matchPattern(op.getOutputPadding(),
+                      m_TorchListOfConstantInts(outputPaddingInts))) {
+      return rewriter.notifyMatchFailure(
+          op, "non-const int output_padding unsupported!");
+    }
+
+    // Canonicalization Logic: Only rewrite if padding provided is 1 element
```
Comment (sahas3) [inline lib/Dialect/Torch/IR/TorchOps.cpp]: padding -> stride/dilation/padding/out_padding or maybe convolution attributes as a general term

**Comment by hariprasadravi** [comment]
@sjarus , @vivekkhandelwal1, @zjgarvey. Please let me know if you have any comments on this PR
